File: src/test/java/net/igsoft/sdi/testclasses/ACreator.java
package net.igsoft.sdi.testclasses;

import net.igsoft.sdi.creator.CreatorBase;
import net.igsoft.sdi.engine.InstanceProvider;
import net.igsoft.sdi.parameter.LaunchType;

public class ACreator extends CreatorBase<AClass, LaunchType> {
    @Override
    public AClass create(InstanceProvider instanceProvider, LaunchType launchType) {
        BClass b = instanceProvider.getOrCreate(BClass.class, launchType);

        return new AClass(b, instanceProvider.getOrCreate(Stepper.class));
    }
}

File: src/test/scala/net/igsoft/sdi/scala/testclasses/ACreator.scala
package net.igsoft.sdi.scala.testclasses

import net.igsoft.sdi.{Creator, InstanceCreator, LaunchType}

class ACreator extends Creator[A, LaunchType] {
  override def create(instanceCreator: InstanceCreator, launchType: LaunchType): A = {
    val b = instanceCreator.getOrCreate(classOf[B], launchType)
    new A(b, instanceCreator.getOrCreate(classOf[Stepper]))
  }
}
